src/commonMain/kotlin/org/kosat/heuristics/Preprocessor.kt

```diff
@@ -0,0 +1,282 @@
+package org.kosat.heuristics
+
+import org.kosat.CDCL
+import org.kosat.Clause
+import org.kosat.VarValue
+import org.kosat.variable
+
+// TODO: add docs and refactoring
+class Preprocessor(private val solver: CDCL) {
```

Comment:
The cognitive complexity of this class is not negligible anymore... Do something with it! Simplify the code, add comments, etc.